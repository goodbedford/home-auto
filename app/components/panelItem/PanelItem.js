import React from "react";
import {Redirect, Route,Link, NavLink} from "react-router-dom";
import Button from "../button/Button";

const PropTypes = React.PropTypes;
const propTypes = {
  isDeleted: PropTypes.bool.isRequired,
  items: PropTypes.array.isRequired,
  match: PropTypes.object.isRequired,
  handleGetRoomClick: PropTypes.func,
  handleDeleteRoom: PropTypes.func
};
const defaultProps = {
  isDeleted: false,
};

const PanelItem = (props) => {
  // console.log("props items", props);
  let nav = null;
  nav = props.items.map((item, index, items) => {
    if(props.match.params.roomId) {
      return (
        <div className="home-auto__panel-group" key={item._id}>
        <NavLink
          onClick={props.handleGetRoomClick(props.match.params.homeId, item._id)}
          exact
          to={{
            pathname:`/homes/${props.match.params.homeId}/rooms/${item._id}`
          }}
          activeClassName="home-auto__panel-item--active"
          className="home-auto__panel-item">
          <span>{item.name}</span>
        </NavLink>
        <NavLink
          onClick={props.handleGetRoomClick(props.match.params.homeId, item._id)}
          to={`/homes/${props.match.params.homeId}/rooms/${item._id}/edit`}
          activeClassName="home-auto__panel-item--active"
          className="home-auto__panel-item home-auto__panel-item--edit-btn">
          <span>
            <i className="fa fa-pencil" aria-hidden="true"></i>
          </span>
        </NavLink>
        <span
          onClick={props.handleDeleteRoom}
          className="home-auto__panel-item home-auto__panel-item--edit-btn">
          <i className="fa fa-trash" aria-hidden="true"></i>
        </span>
      </div>
      );
    }
    else if(props.match.params.homeId) {
      return (
        <NavLink
          exact
          to={{
            pathname: `/homes/${props.match.params.homeId}/rooms/${item._id}`,
          }}
          key={item._id}
          activeClassName="home-auto__panel-item--active"
          className="home-auto__panel-item"><span>{item.name}</span>
        </NavLink>
      );
    }
    else {
      return (
        <NavLink
          exact
          to= {`/homes/${item._id}/rooms`}
          key={item._id}
          activeClassName="home-auto__panel-item--active"
          className="home-auto__panel-item"><span>{item.name}</span>
        </NavLink>
      );
    }
  });

  if(props.isDeleted) {
    return <Redirect to={`/homes/${props.match.params.homeId}/rooms`} />;
  }
  return (
    <div className="home-auto__section-rooms home-auto__section-rooms--bottom-border">
      {nav}
    </div>
  );
};

PanelItem.propTypes = propTypes;
PanelItem.defaultProps = defaultProps;

export default PanelItem;
