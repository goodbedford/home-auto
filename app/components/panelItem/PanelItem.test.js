/* eslint-disable no-undef */

import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import PanelItem from "./PanelItem";

describe("<PanelItem />", () => {
  const rooms = [
    {_id: 123, name: "one room", thermostat: 40, curtains: true, lights: true},
    {_id: 456,name: "2nd room", thermostat: 50, curtains: false, lights: false}
  ];
  const props = {
    isDeleted: false,
    items: rooms,
    match: {params:{}},
    handleGetRoomClick: jest.fn(),
    handleDeleteRoom: jest.fn()
  };
  it("PanelItem snapshot", () => {
    const wrapper = shallow(
      <PanelItem
        {...props}
      />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("PanelItem snapshot with params.homeId", () => {
    props.match = {params: {homeId: 12}};
    const wrapper = shallow(
      <PanelItem
        {...props}
      />);
      // console.log(wrapper.unrendered.props);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("PanelItem snapshot with params.roomId", () => {
    props.match = {params: {homeId: 12, roomId: 456}};
    const wrapper = shallow(
      <PanelItem
        {...props}
      />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
