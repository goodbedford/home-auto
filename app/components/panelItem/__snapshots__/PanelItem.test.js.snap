// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelItem /> PanelItem snapshot 1`] = `
<div
  className="home-auto__section-rooms home-auto__section-rooms--bottom-border"
>
  <NavLink
    activeClassName="home-auto__panel-item--active"
    className="home-auto__panel-item"
    exact={true}
    to="/homes/123/rooms"
  >
    <span>
      one room
    </span>
  </NavLink>
  <NavLink
    activeClassName="home-auto__panel-item--active"
    className="home-auto__panel-item"
    exact={true}
    to="/homes/456/rooms"
  >
    <span>
      2nd room
    </span>
  </NavLink>
</div>
`;

exports[`<PanelItem /> PanelItem snapshot with params.homeId 1`] = `
<div
  className="home-auto__section-rooms home-auto__section-rooms--bottom-border"
>
  <NavLink
    activeClassName="home-auto__panel-item--active"
    className="home-auto__panel-item"
    exact={true}
    to={
      Object {
        "pathname": "/homes/12/rooms/123",
      }
    }
  >
    <span>
      one room
    </span>
  </NavLink>
  <NavLink
    activeClassName="home-auto__panel-item--active"
    className="home-auto__panel-item"
    exact={true}
    to={
      Object {
        "pathname": "/homes/12/rooms/456",
      }
    }
  >
    <span>
      2nd room
    </span>
  </NavLink>
</div>
`;

exports[`<PanelItem /> PanelItem snapshot with params.roomId 1`] = `
<div
  className="home-auto__section-rooms home-auto__section-rooms--bottom-border"
>
  <div
    className="home-auto__panel-group"
  >
    <NavLink
      activeClassName="home-auto__panel-item--active"
      className="home-auto__panel-item"
      exact={true}
      to={
        Object {
          "pathname": "/homes/12/rooms/123",
        }
      }
    >
      <span>
        one room
      </span>
    </NavLink>
    <NavLink
      activeClassName="home-auto__panel-item--active"
      className="home-auto__panel-item home-auto__panel-item--edit-btn"
      to="/homes/12/rooms/123/edit"
    >
      <span>
        <i
          aria-hidden="true"
          className="fa fa-pencil"
        />
      </span>
    </NavLink>
    <span
      className="home-auto__panel-item home-auto__panel-item--edit-btn"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-trash"
      />
    </span>
  </div>
  <div
    className="home-auto__panel-group"
  >
    <NavLink
      activeClassName="home-auto__panel-item--active"
      className="home-auto__panel-item"
      exact={true}
      to={
        Object {
          "pathname": "/homes/12/rooms/456",
        }
      }
    >
      <span>
        2nd room
      </span>
    </NavLink>
    <NavLink
      activeClassName="home-auto__panel-item--active"
      className="home-auto__panel-item home-auto__panel-item--edit-btn"
      to="/homes/12/rooms/456/edit"
    >
      <span>
        <i
          aria-hidden="true"
          className="fa fa-pencil"
        />
      </span>
    </NavLink>
    <span
      className="home-auto__panel-item home-auto__panel-item--edit-btn"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="fa fa-trash"
      />
    </span>
  </div>
</div>
`;
